### PR TITLE
Posts

### PR DESCRIPTION
**Pull request by heimrih**
Post and comments

### CODE REVIEW

**Comment by heimrih** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    if @post.save
+      redirect_to @post
+    else
+      render "new"
+    end
+  end
+
+  def destroy
+    @post = Post.find(params[:id])
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
    @post = Post.find_by id: params[:id]
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+
+    @comment = @post.comments.create(comment_params)
+    redirect_to post_path(@post)
+  end
+
+  def destroy
+    @post = Post.find(params[:post_id])
```
Comment (chan227) [inline app/controllers/comments_controller.rb]:
```suggestion
    @post = Post.find_by id: params[:post_id]
```

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    @comment = @post.comments.find(params[:id])
```
Comment (chan227) [inline app/controllers/comments_controller.rb]:
```suggestion
    @comment = @post.comments.find_by id: params[:id]
```

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    @comment.destroy
```
Comment (chan227) [inline app/controllers/comments_controller.rb]: flash[:success] message should be here

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def edit
+    @post = Post.find(params[:id])
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
    @post = Post.find_by id: params[:id]
```

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def update
+    @post = Post.find(params[:id])
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
    @post = Post.find_by id: params[:id]
```

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    if @post.update(post_params)
+      redirect_to @post
+    else
+      render "edit"
+    end
+  end
+
+  def show
+    @post = Post.find(params[:id])
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
    @post = Post.find_by id: params[:id]
```

File: app/controllers/posts_controller.rb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+    if @post.update(post_params)
+      redirect_to @post
+    else
+      render "edit"
+    end
+  end
+
+  def show
```
Comment (chan227) [inline app/controllers/posts_controller.rb]: If no mistaken, use load_post and place this in private, call it by using before_action

File: config/routes.rb [hunk 9]
```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
   scope "(:locale)", locale: /en/ do
+  resources :posts do
```
Comment (chan227) [inline config/routes.rb]: please rearrange it

**Comment by ngocvt-0484** [comment]
change 2 commits to 1 commit

File: app/views/posts/edit.html.erb [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        <%= msg %>
+      </div>
+    <% end %>
+  <% end %>
+  <p>
+    <%= f.label :title %><br>
+    <%= f.text_field(:title, {:class => "form-control"}) %>
+  </p>
+
+
```
Comment (ngocvt-0484) [inline app/views/posts/edit.html.erb]: remove blank line

File: app/views/posts/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<%= t "posts.add" %>
+<%= form_for :post, url: posts_path do |f| %> <% if @post.errors.any? %> <% @post.errors.full_messages.each do |msg| %>
+<%= msg %>
+<% end %> <% end %>
+
+<%= f.label :title %>
```
Comment (ngocvt-0484) [inline app/views/posts/new.html.erb]: format your html
wrong indent

File: app/views/posts/index.html.erb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+<% @posts.each do |post| %>
+  <div class="well">
+    <h3><%= post.title %></h3>
+    <p><%= post.body %></p>
+    <%= link_to t("read.more"), post_path(post), :class => "btn btn-default" %>
```
Comment (ngocvt-0484) [inline app/views/posts/index.html.erb]:
```suggestion
    <%= link_to t("read.more"), post_path(post), class: "btn btn-default" %>
```

File: app/views/comments/_form.html.erb [hunk 13]
```diff
@@ -0,0 +1,17 @@
+<%= t "comments.add" %>
+<%= form_for([@post, @post.comments.build]) do |f| %>
+  <p>
+    <%= f.label :username %><br>
+    <%= f.text_field(:username, {:class => "form-control"}) %>
```
Comment (ngocvt-0484) [inline app/views/comments/_form.html.erb]:
```suggestion
    <%= f.text_field :username, class: "form-control" %>
```

File: app/helpers/comments_helper.rb [hunk 14]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (ngocvt-0484) [inline app/helpers/comments_helper.rb]: remove empty file

File: app/helpers/posts_helper.rb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
```
Comment (ngocvt-0484) [inline app/helpers/posts_helper.rb]: remove empty file

File: app/controllers/posts_controller.rb [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    @post = Post.find(params[:id])
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    if @post.save
+      redirect_to @post
+    else
+      render "new"
```
Comment (ngocvt-0484) [inline app/controllers/posts_controller.rb]:
```suggestion
      render :new
```

**Comment by heimrih** [comment]
ready

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (chan227) [reply app/controllers/comments_controller.rb]: haven`t changed for this

File: app/controllers/posts_controller.rb [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    if @post.save
+      flash[:success] = t "post.create"
+      redirect_to @post
+    else
+      flash[:warning] = t "post.createfail"
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
      flash.now[:warning] = t "post.createfail"
```

File: app/controllers/posts_controller.rb [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def update
+    @post = Post.find_by id: params[:id]
+
+    if @post.update(post_params)
+      flash[:success] = t "post.update"
+      redirect_to @post
+    else
+      flash[:warning] = t "post.updatefail"
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
      flash.now[:warning] = t "post.updatefail"
```

File: app/controllers/comments_controller.rb [hunk 19]
```diff
@@ -0,0 +1,30 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find_by id: params[:id]
+    @comment = @post.comments.create(comment_params)
```
Comment (chungpv-1008) [inline app/controllers/comments_controller.rb]:
```suggestion
    @comment = @post.comments.create comment_params
```

File: app/controllers/comments_controller.rb [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+    else
+      flash[:warning] = t "comments.deletefail"
+    end
+    redirect_to post_path(@post)
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:username, :body)
```
Comment (chungpv-1008) [inline app/controllers/comments_controller.rb]:
```suggestion
    params.require(:comment).permit :username, :body
```

File: app/controllers/posts_controller.rb [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def edit
+    @post = Post.find_by id: params[:id]
+  end
+
+  def update
+    @post = Post.find_by id: params[:id]
+
+    if @post.update(post_params)
```
Comment (chungpv-1008) [inline app/controllers/posts_controller.rb]:
```suggestion
    if @post.update post_params
```

File: app/controllers/posts_controller.rb [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+      render :edit
+    end
+  end
+
+  def show
+    @post = Post.find_by id: params[:id]
+  end
+
+  def create
+    @post = Post.new(post_params)
```
Comment (chungpv-1008) [inline app/controllers/posts_controller.rb]:
```suggestion
    @post = Post.new post_params
```

File: app/controllers/posts_controller.rb [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+    else
+      flash[:warning] = t "post.deletefail"
+    end
+    redirect_to posts_path
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :body)
```
Comment (chungpv-1008) [inline app/controllers/posts_controller.rb]:
```suggestion
    params.require(:post).permit :title, :body
```

File: app/views/comments/_form.html.erb [hunk 24]
```diff
@@ -0,0 +1,17 @@
+<%= t "comments.add" %>
+<%= form_for([@post, @post.comments.build]) do |f| %>
```
Comment (chungpv-1008) [inline app/views/comments/_form.html.erb]:
```suggestion
<%= form_for [@post, @post.comments.build] do |f| %>
```

File: app/views/comments/_form.html.erb [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+  <p>
+    <%= f.label :username %><br>
+    <%= f.text_field :username, class: "form-control" %>
+  </p>
+
+
+  <p>
+    <%= f.label :body %><br>
+    <%= f.text_area(:body, class: "form-control") %>
```
Comment (chungpv-1008) [inline app/views/comments/_form.html.erb]:
```suggestion
    <%= f.text_area :body, class: "form-control" %>
```

File: app/views/comments/_form.html.erb [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+  </p>
+
+  <p>
+    <%= f.submit(class: "btn btn-default") %>
```
Comment (chungpv-1008) [inline app/views/comments/_form.html.erb]:
```suggestion
    <%= f.submit class: "btn btn-default" %>
```

File: app/views/posts/edit.html.erb [hunk 27]
```diff
@@ -0,0 +1,23 @@
...
+    <%= f.text_field(:title, class: "form-control") %>
+  </p>
+
+  <p>
+    <%= f.label :body %><br>
+    <%= f.text_area(:body, class: "form-control") %>
+  </p>
+
+  <p>
+    <%= f.submit(class: "btn btn-primary") %>
```
Comment (chungpv-1008) [inline app/views/posts/edit.html.erb]:
```suggestion
    <%= f.text_field :title, class: "form-control" %>
  </p>

  <p>
    <%= f.label :body %><br>
    <%= f.text_area :body, class: "form-control" %>
  </p>

  <p>
    <%= f.submit t(".submit"), class: "btn btn-primary" %>
```

File: app/views/posts/new.html.erb [hunk 28]
```diff
@@ -0,0 +1,16 @@
+<%= t "posts.add" %>
+<%= form_for :post, url: posts_path do |f| %>
+  <% if @post.errors.any? %>
+    <% @post.errors.full_messages.each do |msg| %>
+    <%= msg %>
```
Comment (chungpv-1008) [inline app/views/posts/new.html.erb]:
```suggestion
    <% @post.errors.full_messages.each do |msg| %>
      <%= msg %>
    <% end %>
```

File: app/views/posts/new.html.erb [hunk 29]
```diff
@@ -0,0 +1,16 @@
...
+  <% end %>
+<% end %>
+
+<%= f.label :title %>
+<%= f.text_field(:title, {:class => "form-control"}) %>
+
+<%= f.label :body %>
+<%= f.text_area(:body, {:class => "form-control"}) %>
+
+<%= f.submit({:class => 'btn btn-primary'}) %>
```
Comment (chungpv-1008) [inline app/views/posts/new.html.erb]:
```suggestion
<%= f.text_field :title, class: "form-control" %>

<%= f.label :body %>
<%= f.text_area :body, class: "form-control" %>

<%= f.submit t(".submit"), class: "btn btn-primary" %>
```

**Comment by heimrih** [comment]
ready

File: app/controllers/comments_controller.rb [hunk 30]
```diff
@@ -0,0 +1,34 @@
+class CommentsController < ApplicationController
+  def new; end
+
+  def index; end
+
+  def create
```
Comment (saobv-2223) [inline app/controllers/comments_controller.rb]: please, sort as restfult. index -> show -> new -> create -> edit -> update -> destroy

File: app/controllers/comments_controller.rb [hunk 31]
```diff
@@ -0,0 +1,34 @@
...
+    @post = Post.find_by id: params[:id]
+    @comment = @post.comments.create comment_params
+    if @comment.save
+      flash[:success] = t "comments.create"
+    else
+      flash.now[:warning] = t "comments.createfail"
+    end
+    redirect_to post_path(@post)
```
Comment (saobv-2223) [inline app/controllers/comments_controller.rb]: remove ()

File: app/controllers/comments_controller.rb [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def destroy
+    @post = Post.find_by id: params[:id]
+
+    @comment = @post.comments.find params[:id]
```
Comment (saobv-2223) [inline app/controllers/comments_controller.rb]:
```suggestion
    @post = Post.find_by id: params[:id]
    @comment = @post.comments.find params[:id]
```

File: app/controllers/comments_controller.rb [hunk 33]
```diff
@@ -0,0 +1,34 @@
...
+    if @comment.destroy
+      flash[:success] = t "comments.delete"
+    else
+      flash.now[:warning] = t "comments.deletefail"
+    end
+    redirect_to post_path(@post)
```
Comment (saobv-2223) [inline app/controllers/comments_controller.rb]: remove ()

File: app/controllers/posts_controller.rb [hunk 34]
```diff
@@ -0,0 +1,58 @@
...
+      flash[:success] = t "post.create"
+      redirect_to @post
+    else
+      flash.now[:warning] = t "post.createfail"
+      render :new
+    end
+  end
+
+  def destroy
+    @post = Post.find_by id: params[:id]
```
Comment (saobv-2223) [inline app/controllers/posts_controller.rb]: `@post = Post.find_by id: params[:id]` is wrote too many times, you should write in a private method and use before_action for actions that use @post

File: app/views/comments/_comments.html.erb [hunk 35]
```diff
@@ -0,0 +1,12 @@
+<%= t "comments.title" %>
+<% @post.comments.each do |comment| %>
+  <div class="well">
+    <p>
+      <strong>
+        <%= comment.username %>
+      </strong> :
+        <%= comment.body %>
```
Comment (saobv-2223) [inline app/views/comments/_comments.html.erb]:
```suggestion
      <%= comment.body %>
```

File: app/views/comments/_form.html.erb [hunk 36]
```diff
@@ -0,0 +1,17 @@
+<%= t "comments.add" %>
+<%= form_for [@post, @post.comments.build] do |f| %>
+  <p>
+    <%= f.label :username %><br>
+    <%= f.text_field :username, class: "form-control" %>
+  </p>
+
+
```
Comment (saobv-2223) [inline app/views/comments/_form.html.erb]:
```suggestion
  </p>

```

File: app/views/posts/edit.html.erb [hunk 37]
```diff
@@ -0,0 +1,22 @@
...
+    <%= f.label :title %><br>
+    <%= f.text_field :title, class: "form-control" %>
+  </p>
+  <p>
+    <%= f.label :body %><br>
+    <%= f.text_area :body, class: "form-control" %>
+  </p>
+  <p>
+    <%= f.submit t("posts.submit"), class: "btn btn-primary" %>
+
```
Comment (saobv-2223) [inline app/views/posts/edit.html.erb]:
```suggestion
    <%= f.submit t("posts.submit"), class: "btn btn-primary" %>
```

**Comment by heimrih** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 38]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit :title, :body
+  end
+
+  def load_post
+    @post = Post.find_by id: params[:id]
```
Comment (saobv-2223) [inline app/controllers/posts_controller.rb:54]: how about the post not found ???
You should have case if post not found

**Comment by heimrih** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 39]
```diff
@@ -0,0 +1,62 @@
...
+    return if @post.present?
+
+    flash.now[:warning] = t "posts.notfound"
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
    flash[:warning] = t "posts.notfound"
```

File: app/controllers/posts_controller.rb [hunk 40]
```diff
@@ -0,0 +1,62 @@
...
+      flash.now[:warning] = t "posts.updatefail"
+      render :edit
+    end
+  end
+
+  def destroy
+    if @post.destroy
+      flash[:success] = t "posts.delete"
+    else
+      flash.now[:warning] = t "posts.deletefail"
```
Comment (chan227) [inline app/controllers/posts_controller.rb]:
```suggestion
      flash[:warning] = t "posts.deletefail"
```

File: app/controllers/comments_controller.rb [hunk 41]
```diff
@@ -0,0 +1,35 @@
+class CommentsController < ApplicationController
+  def index; end
+
+  def show; end
+
+  def new; end
+
+  def create
+    @post = Post.find_by id: params[:id]
```
Comment (chungpv-1008) [inline app/controllers/comments_controller.rb]: please create a method:
```
def find_post
  @post = Post.find_by id: params[:id]
  return if @post.present?

  flash ...
  redirect ...
end
```
Then call 'before_action' for action 'create' and 'delete'

File: app/controllers/comments_controller.rb [hunk 42]
```diff
@@ -0,0 +1,35 @@
...
+      flash[:success] = t "comments.create"
+    else
+      flash.now[:warning] = t "comments.createfail"
+    end
+    redirect_to post_path @post
+  end
+
+  def destroy
+    @post = Post.find_by id: params[:id]
+    @comment = @post.comments.find params[:id]
```
Comment (chungpv-1008) [inline app/controllers/comments_controller.rb]: I don't understand if you want to use 'params[:id]' for 'Post' or 'Comment' table ?

File: app/views/comments/_comments.html.erb [hunk 43]
```diff
@@ -0,0 +1,12 @@
...
+  <div class="well">
+    <p>
+      <strong>
+        <%= comment.username %>
+      </strong> :
+      <%= comment.body %>
+      <%= link_to t("delete.button"), [comment.post, comment], method: :delete, data:{confirm: t("confirm") } %>
+    </p>
+  </div>
+<% end %>
```
Comment (chungpv-1008) [inline app/views/comments/_comments.html.erb:12]: You should create file '_comment.html.erb', then move code lines to that file, then call that file name back here
```suggestion
render "comment", comment: @post.comments
```

File: app/views/posts/edit.html.erb [hunk 44]
```diff
@@ -0,0 +1,21 @@
+<%= t "posts.edit" %>
+<%= form_for :post, url: post_path(@post), method: :patch do |f| %>
+  <% if @post.errors.any? %>
+    <% @post.errors.full_messages.each do |msg| %>
+      <div class ="alert alert-danger">
+        <%= msg %>
+      </div>
+    <% end %>
+  <% end %>
```
Comment (chungpv-1008) [inline app/views/posts/edit.html.erb]: You should create file '_error_messages.html.erb', then move code lines to that file, then call back here

File: app/views/posts/index.html.erb [hunk 45]
```diff
@@ -0,0 +1,7 @@
+<% @posts.each do |post| %>
+  <div class="well">
+    <h3><%= post.title %></h3>
+    <p><%= post.body %></p>
+    <%= link_to t("read.more"), post_path(post), class: "btn btn-default" %>
+  </div>
```
Comment (chungpv-1008) [inline app/views/posts/index.html.erb:6]: You must create file '_post.html.erb', then move code lines to that file, then call back here

File: app/views/posts/new.html.erb [hunk 46]
```diff
@@ -0,0 +1,14 @@
+<%= t "posts.add" %>
+<%= form_for :post, url: posts_path do |f| %>
+  <% if @post.errors.any? %>
+    <% @post.errors.full_messages.each do |msg| %>
+      <%= msg %>
+  <% end %>
```
Comment (chungpv-1008) [inline app/views/posts/new.html.erb]: similar

File: app/controllers/comments_controller.rb (hunk 42 above)
Comment (heimrih) [reply app/controllers/comments_controller.rb]: I not sure how to create comment for post so i am thinking to find it to show

**Comment by heimrih** [comment]
ready

File: app/controllers/comments_controller.rb [hunk 47]
```diff
@@ -0,0 +1,43 @@
...
+
+  private
+
+  def comment_params
+    params.require(:comment).permit :username, :body
+  end
+
+  def find_post
+    @post = Post.find_by id: params[:id]
+    return if @post.present?
```
Comment (saobv-2223) [inline app/controllers/comments_controller.rb]:
```suggestion
    return if @post
```

File: app/controllers/posts_controller.rb [hunk 48]
```diff
@@ -0,0 +1,62 @@
...
+    else
+      flash.now[:warning] = t "posts.createfail"
+      render :new
+    end
+  end
+
+  def edit; end
+
+
+  def update
```
Comment (saobv-2223) [inline app/controllers/posts_controller.rb]:
```suggestion
  def edit; end

  def update
```

File: app/controllers/posts_controller.rb [hunk 49]
```diff
@@ -0,0 +1,62 @@
...
+
+  private
+
+  def post_params
+    params.require(:post).permit :title, :body
+  end
+
+  def load_post
+    @post = Post.find_by id: params[:id]
+    return if @post.present?
```
Comment (saobv-2223) [inline app/controllers/posts_controller.rb]:
```suggestion
    return if @post
```

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

File: app/controllers/comments_controller.rb [hunk 50]
```diff
@@ -0,0 +1,49 @@
+class CommentsController < ApplicationController
+  before_action :find_post, only: %i(create destroy)
+
+  def index
+    @comment = Comment.all
```
Comment (vanvtt-0952) [inline app/controllers/comments_controller.rb]: plaz add paginate and sort by created at fileds

File: app/controllers/comments_controller.rb [hunk 51]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def show
+    @comment = Comment.find params[:id]
```
Comment (vanvtt-0952) [inline app/controllers/comments_controller.rb]: 1. change from `find` to `find_by`
2. handling null  "@comment` in here

File: app/controllers/comments_controller.rb [hunk 52]
```diff
@@ -0,0 +1,49 @@
...
+    if @comment.save
+      flash[:success] = t "comments.create"
+    else
+      flash.now[:warning] = t "comments.createfail"
+    end
+    redirect_to post_path @post
+  end
+
+  def destroy
+    @comment = @post.comments.find params[:id]
```
Comment (vanvtt-0952) [inline app/controllers/comments_controller.rb]: I think you don't necessary this line
because you had defined find_post method

Comment (heimrih) [reply app/controllers/comments_controller.rb]: image
this error show when i delete the line

**Comment by heimrih** [comment]
ready

File: app/controllers/comments_controller.rb [hunk 53]
```diff
@@ -0,0 +1,45 @@
...
+    if @comment.save
+      flash[:success] = t "comments.create"
+    else
+      flash[:warning] = t "comments.createfail"
+    end
+    redirect_to post_path @post
+  end
+
+  def destroy
+    @comment = @post.comments.find params[:id]
```
Comment (ngocvt-0484) [inline app/controllers/comments_controller.rb]:
```suggestion
    @comment = @post.comments.find_by id: params[:id]
```

File: app/controllers/posts_controller.rb [hunk 54]
```diff
@@ -0,0 +1,60 @@
...
+  before_action :load_post, except: %i(index new create)
+
+  def index
+    @posts = Post.order(:created_at).page params[:page]
+  end
+
+  def show; end
+
+  def new
+    @post = Post.new params[:post]
```
Comment (ngocvt-0484) [inline app/controllers/posts_controller.rb]: dont need params[:post]

**Comment by ngocvt-0484** [comment]
fix conflict plz

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 55]
```diff
@@ -0,0 +1,60 @@
+class PostsController < ApplicationController
+  before_action :load_post, except: %i(index new create)
+
+  def index
+    @posts = Post.order(:created_at).page params[:page]
```
Comment (saobv-2223) [inline app/controllers/posts_controller.rb]:
```suggestion
    @posts = Post.order(:created_at).page(params[:page]).per <number records>
```

File: app/views/posts/index.html.erb [hunk 56]
```diff
@@ -0,0 +1,8 @@
...
+<% end %>
+  <%= paginate @posts %>
```
Comment (saobv-2223) [inline app/views/posts/index.html.erb]:
```suggestion
<%= paginate @posts %>
```

File: app/views/posts/new.html.erb [hunk 57]
```diff
@@ -0,0 +1,10 @@
+<%= t "posts.add" %>
+<%= form_for :post, url: posts_path do |f| %>
+  <%= render "shared/error_postmessages" %>
+
+<%= f.label :title %>
+<%= f.text_field :title, class: "form-control" %>
+<%= f.label :body %>
+<%= f.text_area :body, class: "form-control" %>
+<%= f.submit t("posts.submit"), class: "btn btn-primary" %>
```
Comment (saobv-2223) [inline app/views/posts/new.html.erb]:
```suggestion
  <%= f.label :title %>
  <%= f.text_field :title, class: "form-control" %>
  <%= f.label :body %>
  <%= f.text_area :body, class: "form-control" %>
  <%= f.submit t("posts.submit"), class: "btn btn-primary" %>
```

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

**Pull request merged**